src/build/gen/mod.ts

```diff
@@ -0,0 +1,185 @@
+import { dedent } from "../deps.ts";
+import { dirname, formatPath, isAbsolute, parsePath, relative, resolve } from "../../deps.ts";
+import { autoGenHeader } from "../misc.ts";
+import { Module, hasUserConfigSchema } from "../../project/mod.ts";
+
+export { compileModuleHelper } from "./module.ts";
+export { compileTestHelper } from "./test.ts";
+export { compileScriptHelper } from "./script.ts";
+
+export { compileModuleTypeHelper } from "./module_type.ts";
+export { compileModulePublicUtilsHelper } from "./module_public_utils.ts";
+export { compilePublicUtilsHelpers } from "./public_utils.ts";
+export { compileTypeHelpers } from "./type.ts";
+
+
+function ensureRelativePrefixed(path: string) {
+	if (isAbsolute(path)) return path;
+
+	const parsed = parsePath(path);
+	if (parsed.dir.startsWith(".")) return path;
+	else if (parsed.dir) return formatPath({ ...parsed, dir: `./${parsed.dir}` });
+	else return formatPath({ ...parsed, dir: `.` });
+}
+
+export class GeneratedFile {
```

Comment:
nit: move to separate file

nit: rename to something like `GeneratedCodeBuilder` or something with "Builder" in the name

needs docs on what this class does

cool helper though

Reply:
Moved and renamed, will add jsdoc comments